app/mailers/coupon_mailer.rb

```diff
@@ -0,0 +1,10 @@
+class CouponMailer < ApplicationMailer
+  default from: 'special_coupons@gmail.com'
+
+  def send_coupon_code(user, coupon)
+    @user = user.name
+    @code = coupon.code
```

Comment:
you can set only two instance variables here instead of three

Reply:
Yes sir, fixing.